src/main/scala/sigmastate/lang/SigmaSpecializer.scala

```diff
@@ -130,11 +130,11 @@ class SigmaSpecializer(val builder: SigmaBuilder) {
       val body1 = eval(env + (n -> tagged), body)
       Some(mkMapCollection(col.asValue[SCollection[SType]], tagged.varId, body1))
 
-    case Apply(Select(col,"fold", _), Seq(zero, Lambda(Seq((zeroArg, tZero), (opArg, tOp)), _, Some(body)))) =>
-      val taggedZero = mkTagged(zeroArg, tZero, 21)
-      val taggedOp = mkTagged(opArg, tOp, 22)
-      val body1 = eval(env ++ Seq(zeroArg -> taggedZero, opArg -> taggedOp), body)
-      Some(mkFold(col.asValue[SCollection[SType]], taggedZero.varId, zero, taggedOp.varId, body1))
+    case Apply(Select(col,"fold", _), Seq(zero, Lambda(Seq((accArg, tAccArg), (opArg, tOpArg)), _, Some(body)))) =>
+      val taggedAcc = mkTagged(accArg, tAccArg, 21)
+      val taggedOp = mkTagged(opArg, tOpArg, 22)
+      val body1 = eval(env ++ Seq(accArg -> taggedAcc, opArg -> taggedOp), body)
+      Some(mkFold(col.asValue[SCollection[SType]], taggedOp.varId, zero, taggedAcc.varId, body1))
```

Comment:
@ergomorphic Could you please check if I get the intended parameter sequence right?